airflow/providers/amazon/aws/operators/appflow.py

```diff
@@ -93,6 +98,12 @@ def execute(self, context: Context) -> None:
         self.connector_type = self._get_connector_type()
         if self.flow_update:
             self._update_flow()
+            # previous code had a wait between update and run without explaining why.
+            # since I don't have a way to actually test this behavior,
+            # I'm reproducing it out of fear of breaking workflows.
+            # It might be unnecessary.
```

Comment:
@igorborgest maybe you could and explanation?

Reply:
If I recall correctly, the updates are not propagated atomically. So this sleep was to avoid runs using the previous configuration.

Reply:
checked it with the team, and it's true for on-demand flows (see https://docs.aws.amazon.com/appflow/latest/userguide/flow-triggers.html for a desc of the different types of flows)
I updated the comment accordingly.
Unfortunately, we don't have infra setup to be able to run the appflow system test (because it depends on external sources), so it's a bit hard for me to test it.